File: plugin/integration/scala-plugin/src/scala/org/intellij/plugin/zeppelin/scala/WorksheetUpdateInterpreterHandler.scala
package org.intellij.plugin.zeppelin.scala

import com.intellij.openapi.fileEditor.ex.FileEditorManagerEx
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile
import jetbrains.zeppelin.idea.settings.interpreter.UpdateInterpreterHandler

/**
  * A handler which handle updating of interpreters in worksheet
  */
class WorksheetUpdateInterpreterHandler extends UpdateInterpreterHandler {
  override def updateInterpreter(project: Project): Unit = {
    val currentFile: VirtualFile = FileEditorManagerEx.getInstanceEx(project).getCurrentFile
    if (currentFile == null) return
    val holder = ZeppelinWorksheetWrappersHolder.connectionFor(project)
    holder.updateDependencies(currentFile)
  }
}

File: zeppelin/integration/scala-plugin/src/org/intellij/plugin/zeppelin/scala/WorksheetUpdateInterpreterHandler.kt
package org.intellij.plugin.zeppelin.scala

import com.intellij.openapi.application.runReadAction
import com.intellij.openapi.fileEditor.ex.FileEditorManagerEx
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile
import org.intellij.plugin.zeppelin.extensionpoints.UpdateInterpreterHandler

open class WorksheetUpdateInterpreterHandler : UpdateInterpreterHandler {
    override fun updateInterpreter(project: Project) {
        val currentFile: VirtualFile = runReadAction {
            FileEditorManagerEx.getInstanceEx(project).currentFile
        } ?: return
        val holder = ZeppelinWorksheetWrappersHolder.connectionFor(project)
        holder.updateDependencies(currentFile,force = true)
    }
}